internal/status/conditions_test.go

```diff
@@ -49,3 +53,183 @@ func TestHTTPRouteAddCondition(t *testing.T) {
+		LastTransitionTime: metav1.NewTime(lt),
+	}
+}
+
+func TestComputeGatewayClassAdmittedCondition(t *testing.T) {
+	testCases := []struct {
+		description string
+		valid       bool
+		expect      metav1.Condition
+	}{
+		{
+			description: "valid gatewayclass",
+			valid:       true,
+			expect: metav1.Condition{
+				Type:   string(gatewayapi_v1alpha1.GatewayClassConditionStatusAdmitted),
+				Status: metav1.ConditionTrue,
+				Reason: reasonValidGatewayClass,
+			},
+		},
+		{
+			description: "invalid gatewayclass",
+			valid:       false,
+			expect: metav1.Condition{
+				Type:   string(gatewayapi_v1alpha1.GatewayClassConditionStatusAdmitted),
+				Status: metav1.ConditionFalse,
+				Reason: reasonInvalidGatewayClass,
+			},
+		},
+	}
+
+	for _, tc := range testCases {
+		actual := computeGatewayClassAdmittedCondition(tc.valid)
+		if !apiequality.Semantic.DeepEqual(actual.Type, tc.expect.Type) ||
+			!apiequality.Semantic.DeepEqual(actual.Status, tc.expect.Status) ||
+			!apiequality.Semantic.DeepEqual(actual.Reason, tc.expect.Reason) {
+			t.Fatalf("%q: expected %#v, got %#v", tc.description, tc.expect, actual)
+		}
+	}
+}
+
+func TestConditionChanged(t *testing.T) {
+	testCases := []struct {
+		name     string
+		expected bool
+		a, b     metav1.Condition
+	}{
+		{
+			name:     "nil and non-nil current are equal",
+			expected: false,
+			a:        metav1.Condition{},
+		},
+		{
+			name:     "empty slices should be equal",
+			expected: false,
+			a:        metav1.Condition{},
+			b:        metav1.Condition{},
+		},
+		{
+			name:     "condition LastTransitionTime should be ignored",
+			expected: false,
+			a: metav1.Condition{
+				Type:               string(gatewayapi_v1alpha1.GatewayClassConditionStatusAdmitted),
+				Status:             metav1.ConditionTrue,
+				LastTransitionTime: metav1.Unix(0, 0),
+			},
+			b: metav1.Condition{
+				Type:               string(gatewayapi_v1alpha1.GatewayClassConditionStatusAdmitted),
+				Status:             metav1.ConditionTrue,
+				LastTransitionTime: metav1.Unix(1, 0),
+			},
+		},
+		{
+			name:     "check condition reason differs",
+			expected: true,
+			a: metav1.Condition{
+				Type:   string(gatewayapi_v1alpha1.GatewayConditionReady),
+				Status: metav1.ConditionFalse,
+				Reason: "foo",
+			},
+			b: metav1.Condition{
+				Type:   string(gatewayapi_v1alpha1.GatewayConditionReady),
+				Status: metav1.ConditionFalse,
+				Reason: "bar",
+			},
+		},
+		{
+			name:     "condition status differs",
+			expected: true,
+			a: metav1.Condition{
+				Type:   string(gatewayapi_v1alpha1.GatewayClassConditionStatusAdmitted),
+				Status: metav1.ConditionTrue,
+			},
+			b: metav1.Condition{
+				Type:   string(gatewayapi_v1alpha1.GatewayClassConditionStatusAdmitted),
+				Status: metav1.ConditionFalse,
+			},
+		},
+	}
+
+	for _, tc := range testCases {
+		if actual := conditionChanged(tc.a, tc.b); actual != tc.expected {
+			t.Fatalf("%q: expected %v, got %v", tc.name, tc.expected, actual)
```

Comment:
We've used testify assert.Equal(t, want, got) in other places, might simplify the error message.